app/Console/Commands/BackupDatabase.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Pterodactyl\Console\Commands;
+
+use Illuminate\Console\Command;
+use Symfony\Component\Process\Process;
+use Illuminate\Contracts\Config\Repository as ConfigRepository;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+
+class BackupDatabase extends Command
+{
+    protected $signature = 'db:backup';
+
+    protected $description = 'Provides an exported sql file of the panels current database.';
+
+    /**
+     * @param \Illuminate\Contracts\Config\Repository $config
+     */
+    public function __construct(ConfigRepository $config)
```

Comment:
This should be done in the `handle()` function. This constructor is call whenever a command is executed, so all of this code will be executed (even though technically yes, it does nothing).

Makes more sense to just do it all in the handle. Also, I've moved away from the verbose way of getting configuration values, you're welcome to just use `config()` as needed.